src/component/handlers/edit/setClipboardData.js

```diff
@@ -17,46 +17,59 @@ import type DraftEditor from 'DraftEditor.react';
 var UserAgent = require('UserAgent');
 var setImmediate = require('setImmediate');
 
-var isIE = UserAgent.isBrowser('IE');
+function setClipboardData(e: SyntheticClipboardEvent, editor: DraftEditor, {text, html}): void {
 
-function setClipboardData(e: SyntheticClipboardEvent, editor: DraftEditor, { text, html }): void {
-  if (isIE) {
+  // IE doesn't pass a clipboardData object in the event and instead has a non-standard one attached to window
+  let clipboard = window.clipboardData;
 
-    // We can't always set the clipboard data in IE because of various Group Policies that might be applied.
-    // We need to populate a "copy trap" div with the new HTML and switch focus so the native copy can go
-    // through into that div
-    editor._copyTrap.innerHTML = html;
+  if (clipboard) {
 
-    var selection = window.getSelection();
-    var copyTrapRange = document.createRange();
-    copyTrapRange.selectNodeContents(editor._copyTrap);
-    selection.removeAllRanges();
-    selection.addRange(copyTrapRange);
+    const testClipboardString = 'draft js clipboard set works' + Math.floor(Math.random() * 100000);
+    clipboard.setData('Text', testClipboardString);
 
-    editor._copyTrap.focus();
+    if (clipboard.getData('Text') !== testClipboardString) {
 
-    setImmediate(() => {
-      editor.focus();
-      editor._copyTrap.innerHtml = '';
-    });
+      // We can't always set the clipboard data in IE because of various Group Policies that might be applied.
+      // We need to populate a "copy trap" div with the new HTML and switch focus so the native copy can go
+      // through into that div
+      editor.setMode('copy');
+      editor._copyTrap.innerHTML = html;
 
-    // Let the native event go through
-    return;
-  }
+      var selection = window.getSelection();
+      var originalRange = selection.getRangeAt(0).cloneRange();
```

Comment:
This should check to see if there is a range in the first place. There should be... but you never know when it comes to browsers.